.github/actions/get-version-number-and-issue-url-v1/action.yml

```diff
@@ -0,0 +1,51 @@
+name: get-version-number-and-issue-url
+description: 'A small GitHub Action to get the version number and issue URL for a release candidate'
+
+inputs:
+  disable-get-issue-url:
+    description: 'Whether or not to get the issue URL for the release candidate'
+    default: 'false'
+    required: false
+
+outputs:
+  version-number:
+    description: 'The version number for the release candidate'
+    value: ${{ steps.get_version_number.outputs.version-number }}
+  issue-url:
+    description: 'The issue URL for the release candidate'
+    value: ${{ steps.get_issue_url.outputs.issue-url }}
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        # Fetch all history
+        fetch-depth: 0
+
+    - name: Get release-candidate issue URL
+      if: ${{ inputs.disable-get-issue-url == 'false' }}
+      id: get_issue_url
+      shell: bash
+      run: |
+        issueUrl=$(gh issue list --repo ${{ github.repository }} --label release --state open --json url --jq '.[0].url')
+        echo "Issue URL: $issueUrl"
+
+        echo "issue-url=$issueUrl" >> $GITHUB_OUTPUT
+
+    - name: Get release-candidate version number
+      id: get_version_number
+      shell: bash
+      run: |
+        pullRequestID=$(gh pr list --repo ${{ github.repository }} --json number -s merged --base release --jq '.[0].number')
```

Comment:
The issue you get in the `Get release-candidate issue URL` step should also include the version number in the title. It would be easier to get the version number from there rather than making an additional 2 calls to pull requests